### DISCUSSION TITLE
Uso de Arrays y Modularización.

### DISCUSSION THREAD

**Discussion by ferzen1209** [IIC2133-PUC/2024-2#40] [Tarea 0] [answered]
Hola, me preguntaba si se podían utilizar arrays dinamicos?, ya que no he vista en ninguna parte que diga que no, además dado que las variables K, P, E se definen dentro dentro de main() y a su vez main() no admite la creacion de funciones(modularización) dentro de si, se hace muy complicado la modularización sin utilizar arrays dinamicos o sin caer en malas practicas como por ejemplo reservar mucha memoria para un array esperando que K no sea mayor . En caso de que no, hay alguna recomendación para resolver este problema?

**Comment by sebacarrascop** [comment, marked as answer]
Hola!
Como es de esperar, dado que no se ven arreglos dinamicos en este curso no se espera que se usen, sin embargo, como dices 

> ya que no he vista en ninguna parte que diga que no

puedes hacerlo si consideras que es un aporte a tu tarea.

Y dado que no puedes (idealmente) crear funciones dentro del main, una alternativa es crear las funciones fuera del main y llamarlas desde el main.

Si tienes dudas de como hacer esto te recomiendo ver el [modulo #10 del taller intro a C](https://github.com/DCCentral-de-Apuntes/intro-C/blob/master/Capsulas/10%20-%20Modulos/10%20-%20Modulos.pdf)

Si algo no quedo claro no dudes en seguir preguntando, saludos!
